src/extension/content-script/index.ts

```diff
@@ -0,0 +1,110 @@
+import browser from "webextension-polyfill";
+import type { MessageWebLNWithOrigin, WebLNEventData } from "~/types";
+
+import extractLightningData from "./batteries";
+import injectScript from "./injectScript";
+import getOriginData from "./originData";
+import shouldInject from "./shouldInject";
+
+// WebLN calls that can be executed from the WebLNProvider.
+// Update when new calls are added
+export const weblnCalls = [
+  "enable",
+  "getInfo",
+  "lnurl",
+  "sendPaymentOrPrompt",
+  "keysendOrPrompt",
+  "makeInvoice",
+  "signMessageOrPrompt",
+  "verifyMessage",
+] as const;
+// calls that can be executed when webln is not enabled for the current content page
+const disabledCalls = ["enable"];
+
+let isEnabled = false; // store if webln is enabled for this content page
+let callActive = false; // store if a webln is currently active. Used to prevent multiple calls in parallel
+
+async function init() {
+  const inject = await shouldInject();
+  if (!inject) {
+    return;
+  }
+
+  injectScript(); // injects the webln object
+
+  // extract LN data from websites
+  browser.runtime.onMessage.addListener((request) => {
+    if (request.action === "extractLightningData") {
+      extractLightningData();
+    }
+  });
+
+  // message listener to listen to inpage webln calls
+  // those calls get passed on to the background script
+  // (the inpage script can not do that directly, but only the inpage script can make webln available to the page)
+  window.addEventListener(
+    "message",
+    (ev: { source: MessageEvent["source"]; data: WebLNEventData }) => {
+      // Only accept messages from the current window
+      if (ev.source !== window) {
+        return;
+      }
+
+      // QUESTION: when does happen @bumi?: ev.data.response
```

Comment:
@bumi

Reply:
I think i.e. here:
https://github.com/getAlby/lightning-browser-extension/blob/feat/1216_refactor-prompt-popup-lnurl-handling--lnurlauth/src/extension/background-script/actions/lnurl/withdraw.ts#L14

Reply:
I don't think this should be touched here. We initially wanted to change the way lnurl-* routing works that people can also copy&paste the lnurl-pay string. 
Now it is touching all the routes and here now even the content-script and how messaging between inpage<->content<->background is done. (which is quite complex)

Reply:
You mean this file in general or something specific? Lisa only added some types and switched it to `.ts`.
There's still some `FixMe/any` but step by step this will help us to make this easier to understand and keep i.e. actions aligned throughout the app.

Reply:
The file `content-script` is only converted to TS.
No functionality / code was changed. 
The only difference is that some types are added to let TypeScript understand the file.

Reply:
I do not see here in the diff what was changed. and most bugs start with "only added" :) 
but I'd just want you to be aware that this is a critical piece and this initial rather small refactoring got a bit big and bigger when we throw in some ts fixes here and there. (I know we said we switch to TS when we touch things, but this should not be touched here, and ev.data.response is out of scope here)

Reply:
No diff because file-ending was changed.
Attached a diff file which shows changes without changing the file-ending.
[diff.txt](https://github.com/getAlby/lightning-browser-extension/files/9296075/diff.txt)

Reply:
The commit history is very detailed to make a review more easy.
If you go through the commits, then you can see step by step what changed in this file too 
https://github.com/getAlby/lightning-browser-extension/pull/1257/commits

(useful Keyboard commands "n" and "p" to browse through the commits)